lib/blueprinter/base.rb

```diff
@@ -153,10 +153,10 @@ def self.association(method, options = {}, &block)
 
       current_view << Association.new(
         method: method,
-        name: options.delete(:name) || method,
-        extractor: options.delete(:extractor) || AssociationExtractor.new,
-        blueprint: options.delete(:blueprint),
-        view: options.delete(:view) || :default,
+        name: options.fetch(:name) { method },
+        extractor: options.fetch(:extractor) { AssociationExtractor.new },
+        blueprint: options.fetch(:blueprint),
+        view: options.fetch(:view, :default),
         options: options.merge(block: block)
```

Comment:
To preserve behavior here, we may want to exclude the keys that were previously deleted from the `options` hash (even though `Association` merges `view` and `blueprint` back in):

```suggestion
        options: options.except(
          :name,
          :extractor,
          :blueprint,
          :view 
        ).merge(block: block)
```